Update site for Isle of Dread and Vecna Unleashed

tevwoods opened the pull request:
Hello there!

I love your website and it has been extremely helpful in figuring out my endgame builds for the last couple of characters.

In a few cases I've run into difficulty with Dino Bone Crafting and the Lost Purpose sets from Vecna Unleashed not being implemented, which makes it difficult to keep track of things.

I took the liberty of completing your implementation of Dino Bone crafting, and adding my own implementation for the Lost Purpose sets for my own use, and thought I would offer my code up if you want to incorporate it into the website for others to use. I also made a few tweaks so that Set Augments will parse correctly and their set and set bonus will be tracked appropriately in the site.

In addition, I have fixed a couple of bugs:

- Fixed: Set bonuses that gave a bonus to Well Rounded or All Ability Scores were not applying their bonus to individual Ability score affixes that were marked as important. Now set affixes will be checked to see if they are a group affix that might apply to multiple affixes.

- Fixed: There were instances of "Well-Rounded" in addition to "Well Rounded", so I added a synonym for that.

- Fixed: The parser was not set up to call "parse_quests" so Raid items were not updating correctly because the list of raid quests was outdated. Added a call to parse_quests in the main parse method.

Data:

I went ahead and added all (I think, I certainly may have missed some) of the augments that are currently available to the `augments.xlsx` spreadsheet.

tevwoods commented on data-builder/build.py:
```diff
@@ -33,6 +34,8 @@ def build_data(clearCache, discordURL):
 
     parse_item_types()
+
+    parse_quests()
```
This was a small change to call the `parse_quests` method you had already created so that the parser will continue to update the raid quest list.

tevwoods commented on data-builder/lost_purpose.json:
Given that this was a pretty small data set, I thought it would be quicker to do the data by hand than to use parsing for this

tevwoods commented on data-builder/parse_augments.py:
```diff
@@ -29,14 +29,19 @@ def parse_augments():
             level = row[0].value
             value = row[3].value
             bonusType = row[4].value
+            description = row[5].value
```
I started to incorporate the notion of a "description" to go along with each affix, but I was not yet able to get it fully connected up throughout the application. I think it would be awesome if the site could show tooltips when you hover over terms that describes what they do.

tevwoods commented on data-builder/parse_augments.py:
```diff
@@ -50,6 +55,11 @@ def parse_augments():
                 if augmentName:
                     option['name'] = augmentName
 
+                if 'name' in option and option['name'].startswith("Set Augment: "):
```
This supports the Set Augments having an actual `set` property on their option, so when added to an item, the set bonus will work correctly.

tevwoods commented on data-builder/parse_dinosaur_bone_crafting.py:
```diff
@@ -29,24 +32,57 @@ def get_systems_from_page(soup):
         body = table.find('tbody')
 
-        system_name = headers[0].getText()
+        system_name = headers[0].getText().rstrip()
```
These items were coming back with a newline character at the end, which was causing a mismatch later.

tevwoods commented on data-builder/parse_dinosaur_bone_crafting.py:
```diff
@@ -29,24 +32,57 @@ def get_systems_from_page(soup):
 
         systems[system_name] = {'*': []}
 
+        create_slot_specific_systems(system_name, systems)
```
In order to facilitate the fact that some dino augments give different benefits depending upon where they are slotted, I added unique slot names for these "alternate" items.

tevwoods commented on site/package.json:
```diff
@@ -29,7 +29,7 @@
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "^16.2.5",
+    "@angular-devkit/build-angular": "13.3.11",
```
I had to make this change or the build instructions in readme just didn't work

tevwoods commented on site/src/app/equipped.service.ts:
```diff
@@ -137,7 +137,7 @@ export class EquippedService {
         if (item.crafting) {
           for (const crafting of item.crafting) {
-            if (crafting.selected.affixes.length) {
+            if (crafting.selected.affixes.length || crafting.selected.set) {
```
Set augments don't have affixes, but they do have a set property

illusionistpm commented:
Thanks so much for this. I haven't been playing a ton lately and definitely not in end game stuff so it has languished.

The PR is conflicting at the moment because the nightly build job updated the asset files. Please revert your changes to `site/src/assets` and I'll be happy to approve & merge it in.

tevwoods commented:
I certainly know how that goes! I'm just glad I could help out.

I removed the conflicting files, added a few more augments I missed on the first pass, and fixed a bug I had introduced late in a refactor.  Should be good to go now, let me know if you need me to tweak anything

illusionistpm commented:
Thanks again! I'm thrilled to have someone else contributing to the site. 

The site changes will be live shortly and the asset changes will be updated tonight.
